colorizer_data/bin/example_scripts/convert_emt_h2b_labelfree.py

```diff
@@ -296,25 +378,30 @@ def make_features(
         outliers=outliers,
     )
 
-    for info in FEATURE_INFO:
-        data = dataset[info.column_name].to_numpy()
-        writer.write_feature(data, info)
-
-    # Custom: Write a categorical feature based on cell type by aggregating the masks
-    # 0 = colony cell, 1 = edge cell, 2 = migratory cell
-    migratory_mask = dataset["Migratory Cell (colony mask)"].to_numpy()
-    edge_mask = dataset["Edge cell (colony mask)"].to_numpy()
-    cell_types = np.zeros(shape=edge_mask.shape)
-    cell_types += edge_mask
-    cell_types += migratory_mask * 2
-    writer.write_feature(
-        cell_types,
-        FeatureInfo(
-            label="Cell type",
-            type=FeatureType.CATEGORICAL,
-            categories=["Colony Cell", "Edge Cell", "Migratory Cell"],
-        ),
-    )
+    for feature_column in FEATURE_COLUMNS:
+        info = None
+        data = dataset[feature_column].to_numpy()
+
+        if feature_column in FEATURE_INFO_OVERRIDES:
+            info = FEATURE_INFO_OVERRIDES[feature_column]
+            info.column_name = feature_column
+            writer.write_feature(data, info)
+        else:
+            # Get default feature info
+            (label, units) = extract_units_from_feature_name(feature_column)
+            if units == "um":
+                units = "µm"
+            info = FeatureInfo(
+                label=label,
+                unit=units,
+                column_name=feature_column,
+                type=FeatureType.CONTINUOUS,
+            )
+            # Auto-detect categorical features
+            if data.dtype.kind in {"U", "S", "O"}:  # unicode, string, or object
```

Comment:
seems like this logic should just be inside of write_feature

Reply:
Moved it in #30! The changes were sizeable enough (and needed unit tests) so they're now in their own PR. I'll merge them all in a stack when everything is approved.